backend.native/tests/codegen/inlineClass/nestedInlineClasses.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2010-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license
+ * that can be found in the LICENSE file.
+ */
+
+package codegen.inlineClass.nestedInlineClasses
+
+import kotlin.test.*
+
+interface I {
+    inline class IC(val s: String)
+}
+
+interface I2 {
+    inline class IC(val s: String)
+}
+
+@Test fun runTest() {
+    assertEquals("zzz", I.IC("zzz").s)
+    assertEquals("zzz", I2.IC("zzz").s)
```

Comment:
The test doesn't involve boxing, so it might leave a box/unbox functions clash undetected.